archai/nlp/train.py

```diff
@@ -857,16 +864,22 @@ def create_or_load_model(args, device, ntokens)->Tuple[ArchaiModel, dict]:
     if args.qat and not args.pretrained_path:
         logging.warning('QAT usually starts from a pretrained model. Check the --pretrained_path argument.')
 
+    if args.qat and args.mixed_qat:
+        raise ValueError('QAT and Mixed QAT cannot be used at the same time.')
+
     if args.pretrained_path:
         logging.info('Overwriting the provided model config with the pretrained model config.')
-        model, model_config, checkpoint = ArchaiModel.load_model(AVAILABLE_MODELS[args.model_type], args.pretrained_path, on_cpu=False)
+        model, model_config, checkpoint = ArchaiModel.load_model(AVAILABLE_MODELS[args.model_type], args.pretrained_path, args, on_cpu=False)
     else:
         model_cls = AVAILABLE_MODELS[args.model_type]
         model = model_cls(**model_config)
 
     if args.qat:
         model = prepare_with_qat(model, onnx_compatible=True)
 
+    if args.mixed_qat:

```

Comment:
What happens if we have the following:

Whenever we instance a `MemTransformerLM` class, the `self` is already the module itself, while when we instance a HF model, such as `HfGPT2`, the module is inside `self.model`.

Both cases works when using the MixedQAT?

Reply:
Yes, Pytorch treats the same way both cases. HfGPT2 is itself a module (Module->ArchaiModel->HfGPT2).